src/Channels.Networking.Windows.RIO/Internal/RioThread.cs

```diff
@@ -120,7 +126,10 @@ private static void OnThreadStart(object state)
                             RioTcpConnection connection;
                             if (thread._connections.TryGetValue(result.ConnectionCorrelation, out connection))
                             {
-                                connection.Complete(result.Status, result.RequestCorrelation, result.BytesTransferred);
+                                ThreadPool.QueueUserWorkItem(
```

Comment:
The whole ProcessRequest async and user code runs on the RIO thread currently; and as its bypassing buffering in Winsock any lag will cause it to start dropping packets for all of the requests on that thread...